AudiusProject/audius-protocol
# Make storagePath metrics work in cluster mode

### Description
When in cluster mode, this PR allows the primary to record metrics by sending a message to the special worker so the special worker can record it on the primary's behalf. This is needed because in cluster mode the primary doesn't record any of its own metrics. It instead aggregates metrics from the workers.

This PR also moves a few untyped state machine utils into a more appropriate prometheus directory, types them, and splits them utils into the utils used during setup and after setup to avoid circular imports.

This only impacts storagePath metrics because they run on the primary instead of a worker. The storagePath metrics are:
- `audius_cn_files_migrated_from_legacy_path`
- `audius_cn_files_migrated_from_custom_path`


### Tests
1. Create a user with a track: `A up; A seed clear; A seed create-user; A seed upload-track`
2. Manually change their files to have a legacy path: `UPDATE "Files" SET "storagePath" = CONCAT('/file_storage', SUBSTRING("storagePath" FROM 20));`
3. Wait for pagination to process one of their files, and verify that `/prometheus_metrics` shows the storagePath metrics having non-empty values. Note that it's expected to fail in the screenshot because I only changed the db so it didn't have any file to copy from the legacy path on disk.
<img width="1314" alt="Screen Shot 2022-12-12 at 1 18 23 PM" src="https://user-images.githubusercontent.com/4657956/207156707-064f6db2-5adc-4c98-ba69-186e64c43312.png">


### Monitoring - How will this change be monitored? Are there sufficient logs / alerts?
Look for logs containing the messages "Unexpected metric type" or "Error recording metric." We'll also want a Grafana panel to make sure these metrics are trending downward after Service Providers update.
